manifests-gen/providers.go

```diff
@@ -0,0 +1,165 @@
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	powerVSProvider         = "powervs"
+	ibmCloudProvider        = "ibmcloud"
+	metadataFilePath        = "./metadata.yaml"
+	kustomizeComponentsPath = "./config/default"
+)
+
+type provider struct {
+	Name       string                    `json:"name"`
+	PType      clusterctlv1.ProviderType `json:"type"`
+	Version    string                    `json:"version"`
+	components repository.Components
+	metadata   []byte
+}
+
+// loadComponents loads components from the given provider.
+func (p *provider) loadComponents() error {
+	// Create new clusterctl config client
+	configClient, err := configclient.New("")
+	if err != nil {
+		return err
+	}
+
+	// Create new clusterctl provider client
+	providerConfig, err := configClient.Providers().Get(p.Name, p.PType)
+	if err != nil {
+		return err
+	}
+
+	// Set options for yaml processor
+	options := repository.ComponentsOptions{
+		TargetNamespace:     targetNamespace,
+		SkipTemplateProcess: true,
+	}
+
+	// Compile assets using kustomize.
+	rawComponents, err := fetchAndCompileComponents(path.Join(projDir, kustomizeComponentsPath))
+	if err != nil {
+		return err
+	}
+
+	// Ininitialize new clusterctl repository components, this should some yaml processing
+	p.components, err = repository.NewComponents(repository.ComponentsInput{
+		Provider:     providerConfig,
+		ConfigClient: configClient,
+		Processor:    yamlprocessor.NewSimpleProcessor(),
+		RawYaml:      rawComponents,
+		Options:      options,
+	})
+	if err != nil {
+		return err
+	}
+
+	content, err := os.ReadFile(path.Join(projDir, metadataFilePath))
+	if err != nil {
+		return fmt.Errorf("error while reading metadata file: %w", err)
+	}
+
+	p.metadata = content
+
+	return err
+}
+
+func (p *provider) providerTypeName() string {
+	return strings.ReplaceAll(strings.ToLower(string(p.PType)), "provider", "")
+}
+
+func (p *provider) writeProviderComponentsConfigmap(fileName string, objs []unstructured.Unstructured) error {
+	combined, err := utilyaml.FromUnstructured(objs)
+	if err != nil {
+		return err
+	}
+
+	annotations := openshifAnnotations
+	annotations[techPreviewAnnotation] = techPreviewAnnotationValue
+
+	cm := &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      p.Name,
+			Namespace: targetNamespace,
+			Labels: map[string]string{
+				"provider.cluster.x-k8s.io/name":    p.Name,
+				"provider.cluster.x-k8s.io/type":    p.providerTypeName(),
+				"provider.cluster.x-k8s.io/version": p.Version,
+			},
+			Annotations: annotations,
+		},
+		Data: map[string]string{
+			"metadata":   string(p.metadata),
+			"components": string(combined),
+		},
+	}
+
+	cmYaml, err := yaml.Marshal(cm)
+	if err != nil {
+		return err
+	}
+
+	return os.WriteFile(path.Join(*manifestsPath, fileName), ensureNewLine(cmYaml), 0600)
+}
+
+func importProvider(p provider) error {
+	fmt.Printf("Processing provider %s\n", p.Name)
```

Comment:
Debug ?

Reply:
Same as https://github.com/openshift/cluster-capi-operator/pull/121#discussion_r1405973164

Reply:
As I read it, I think that given this is used as a CLI tool, I think the author had the intent for this to print to stdout the processing of the various components. I'm fine to change it to a log.Print if you prefer that though.

Reply:
Yeah i think change these to log messages rather than `fmt.Printf`